code/modules/projectiles/updated_projectiles/gun_helpers.dm

```diff
@@ -739,10 +739,18 @@ should be alright.
 		A.ui_action_click(usr, src)
 
 
+/obj/item/weapon/gun/verb/toggle_rail_attachment()
+	set category = "Weapons"
+	set name = "Toggle Rail Attachment"
+	set desc = "Uses the rail attachement currently attached to the gun."
+
+	if(!usr)
+		return
 
+	rail.activate_attachment(src, usr)
```

Comment:
```suggestion
	rail?.activate_attachment(src, usr)
```
Else if you use this verb on a gun without a rail attachment a runtime will occur.